developmentseed/observe-api
# Feature/badge api

* Creates a run script to calculate badges for all users
* Contains `numObservations` sample metric: get all users that have had 'threshold' observations and the time at which they achieved the badge
* Fixes bug in `fetchObservations`
* Fixes seed to be a little more random

To test (after migrating)
1. Make sure to have observations in the database with new dates (run `seed-dev` again)
2. Add a badge in the badges table with the following properties:
```
id: 1
name: Pioneer
description: {"metric": "numObservations", "threshold": 10 }
```
3. Run the `calculate-badges` npm script
4. Check that `badges_users` has new entries
